File: app/src/main/java/com/example/chatapp/EmojiAdapter.java
package com.example.chatapp;

import android.content.Context;
import android.net.Uri;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class EmojiAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>
{
    Context context;
    ArrayList<Integer> emojiName;
    private OnEmojiListener mClickListener;
    public EmojiAdapter( ArrayList<Integer> emojiName)
    {
        this.emojiName=emojiName;
    }
    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
    {
        context=parent.getContext();
        LayoutInflater inflater=LayoutInflater.from(parent.getContext());
        View cell=inflater.inflate(R.layout.emoji_cell,parent,false);
        Item item=new Item(cell);
        return item;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, final int position)
    {
        GlideApp
                .with(context)
                .load(emojiName.get(position))
                .into(((Item)holder).emojiView);
        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(mClickListener!=null)
                {
                    mClickListener.onEmojiClick(position);
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return emojiName.size();
    }
    public class Item extends RecyclerView.ViewHolder
    {
        ImageView emojiView;
        public Item(View itemView)
        {
            super(itemView);
            emojiView=(ImageView)itemView.findViewById(R.id.emojiView);
        }
    }

    public interface OnEmojiListener
    {
        public void onEmojiClick(int position);
    }
    public void setEmojiListener(EmojiAdapter.OnEmojiListener mClickListener)
    {
        this.mClickListener=mClickListener;
    }
}

File: app/src/main/java/com/example/chatapp/EmojiAdapter.kt
package com.example.chatapp

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import java.util.*

class EmojiAdapter(var emojiName: ArrayList<Int>) : RecyclerView.Adapter<ViewHolder>() {
    var context: Context? = null
    private var mClickListener: OnEmojiListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        val inflater = LayoutInflater.from(parent.context)
        val cell = inflater.inflate(R.layout.emoji_cell, parent, false)
        return Item(cell)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        GlideApp
                .with(context!!)
                .load(emojiName[position])
                .into((holder as Item).emojiView)
        holder.itemView.setOnClickListener {
            if (mClickListener != null) {
                mClickListener!!.onEmojiClick(position)
            }
        }
    }

    override fun getItemCount(): Int {
        return emojiName.size
    }

    inner class Item(itemView: View) : ViewHolder(itemView) {
        var emojiView: ImageView

        init {
            emojiView = itemView.findViewById<View>(R.id.emojiView) as ImageView
        }
    }

    interface OnEmojiListener {
        fun onEmojiClick(position: Int)
    }

    fun setEmojiListener(mClickListener: OnEmojiListener?) {
        this.mClickListener = mClickListener
    }

}